src/Scheduler.ino

```diff
@@ -422,105 +416,66 @@ void process_plugin_task_timer(unsigned long id) {
-
-//timer_data.TaskIndex        = deviceIndex;
-  timer_data.Par1             = Par1;
-  timer_data.Par2             = Par2;
-  timer_data.Par3             = Par3;
-  timer_data.Par4             = Par4;
-  timer_data.Par5             = Par5;
-  systemTimers[systemTimerId] = timer_data;
-  setTimer(PLUGIN_TIMER, systemTimerId, msecFromNow);
-}
-
-void process_plugin_timer(unsigned long id) {
-  START_TIMER;
-  const systemTimerStruct timer_data = systemTimers[id];
-  struct EventStruct TempEvent;
-//  TempEvent.TaskIndex = timer_data.TaskIndex;
-
-// extract deviceID from timer id:
-  const deviceIndex_t deviceIndex = ((1 << 8) -1) & id;
-
-  TempEvent.Par1      = timer_data.Par1;
-  TempEvent.Par2      = timer_data.Par2;
-  TempEvent.Par3      = timer_data.Par3;
-  TempEvent.Par4      = timer_data.Par4;
-  TempEvent.Par5      = timer_data.Par5;
-
-  // TD-er: Not sure if we have to keep original source for notifications.
-  TempEvent.Source = VALUE_SOURCE_SYSTEM;
-//  const deviceIndex_t deviceIndex = getDeviceIndex_from_TaskIndex(timer_data.TaskIndex);
-
-  /*
-     String log = F("proc_system_timer: Pluginid: ");
-     log += deviceIndex;
-     log += F(" taskIndex: ");
-     log += timer_data.TaskIndex;
-     log += F(" sysTimerID: ");
-     log += id;
-     addLog(LOG_LEVEL_INFO, log);
-   */
-  systemTimers.erase(id);
-
-  if (validDeviceIndex(deviceIndex)) {
-    String dummy;
-    Plugin_ptr[deviceIndex](PLUGIN_ONLY_TIMER_IN, &TempEvent, dummy);
-  }
-  STOP_TIMER(PROC_SYS_TIMER);
-}
-
 /*********************************************************************************************\
 * GPIO Timer
 * Special timer to handle timed GPIO actions
 \*********************************************************************************************/
-unsigned long createGPIOTimerId(byte pinNumber, int Par1) {
+unsigned long createGPIOTimerId(byte GPIOType, byte pinNumber, int Par1) {
   const unsigned long mask  = (1 << TIMER_ID_SHIFT) - 1;
-  const unsigned long mixed = (Par1 << 8) + pinNumber;
+//  const unsigned long mixed = (Par1 << 8) + pinNumber;
+  const unsigned long mixed = (Par1 << 16) + (pinNumber << 8) + GPIOType;
 
   return mixed & mask;
 }
 
-void setGPIOTimer(unsigned long msecFromNow, int Par1, int Par2, int Par3, int Par4, int Par5)
+void setGPIOTimer(unsigned long msecFromNow, int GPIOType, int Par1, int Par2, int Par3, int Par4, int Par5)
 {
-  // Par1 & Par2 form a unique key
-  const unsigned long systemTimerId = createGPIOTimerId(Par1, Par2);
+  // Par1 & Par2 & GPIOType form a unique key
+  const unsigned long systemTimerId = createGPIOTimerId(GPIOType, Par1, Par2);
 
   setTimer(GPIO_TIMER, systemTimerId, msecFromNow);
 }
 
 void process_gpio_timer(unsigned long id) {
-  // FIXME TD-er: Allow for all GPIO commands to be scheduled.
-  byte pinNumber     = id & 0xFF;
-  byte pinStateValue = (id >> 8);
-
-  digitalWrite(pinNumber, pinStateValue);
+  byte GPIOType      = id & 0xFF;
+  byte pinNumber     = (id >> 8);
```

Comment:
I wonder why the compiler is not sending out a warning when you shift without bit masking.
I know the old code does this, but better make it explicit:
```c++
byte pinNumber = static_cast<byte>((id >> 8) & 0xFF);
```

Reply:
only pinNumber should change? Or also GPIOType and/or pinStateValue?

Reply:
The other one also. I just added it as an example.

Reply:
you mean like this?
byte GPIOType = static_cast<byte>((id) & 0xFF);
byte pinNumber = static_cast<byte>((id >> 8) & 0xFF);
byte pinStateValue = static_cast<byte>((id >> 16) & 0xFF);

Reply:
```c++
byte GPIOType = static_cast<byte>((id) & 0xFF);
byte pinNumber = static_cast<byte>((id >> 8) & 0xFF);
byte pinStateValue = static_cast<byte>((id >> 16) & 0xFF);
```

You forgot the `<byte>` for the static_cast

Reply:
done